pyomo/core/base/component.py

```diff
@@ -1205,6 +1432,8 @@ def parse_cuid(self, label):
         TODO
         """
         cList = label.split('.')
+        # NOTE: This split is not safe for labels that include
+        # decimal indices.
```

Comment:
CUIDs don't work with decimal indices due (at least in part) to this `split`.

Reply:
This definitely seems like a problem. If we don't fix this in this PR make sure to open a separate issue for it.